doc/py_tutorials/py_setup/py_setup_in_ubuntu/py_setup_in_ubuntu.markdown

```diff
@@ -0,0 +1,242 @@
+2.4.5 while latest OpenCV version is 2.4.6. With respect to Python API, latest version will always
+contain much better support. Also, there may be chance of problems with camera support, video
+playback etc depending upon the drivers, ffmpeg, gstreamer packages present etc.
+
+So my personal preference is next method, i.e. compiling from source. Also at some point in time,
+if you want to contribute to OpenCV, you will need this.
+
+Installing OpenCV from source
+-----------------------------
+
+Compiling from source may seem a little complicated at first, but once you succeeded in it, there is
+nothing complicated.<br>
+
+First we will install some dependencies. Some are compulsory, some are optional. Optional
+dependencies, you can leave if you don't want.<br>
+
+### Compulsory Dependencies
+
+We need **CMake** to configure the installation, **GCC** for compilation, **Python-devel** and
+**Numpy** for creating Python extensions etc.<br>
+
+`sudo apt-get install cmake`<br>
+`sudo apt-get install python-devel numpy`<br>
+`sudo apt-get install gcc gcc-c++`<br>
+
+Next we need **GTK** support for GUI features, Camera support (libdc1394, libv4l), Media Support
+(ffmpeg, gstreamer) etc.<br>
+
+`sudo apt-get install gtk2-devel`<br>
+`sudo apt-get install libdc1394-devel`<br>
+`sudo apt-get install libv4l-devel`<br>
+`sudo apt-get install ffmpeg-devel`<br>
+`sudo apt-get install gstreamer-plugins-base-devel`<br>
+
+### Optional Dependencies
+
+Above dependencies are sufficient to install OpenCV in your ubuntu machine. But depending upon your
+requirements, you may need some extra dependencies. A list of such optional dependencies are given
+below. You can either leave it or install it, your call :)
+
+OpenCV comes with supporting files for image formats like PNG, JPEG, JPEG2000, TIFF, WebP etc. But
+it may be a little old. If you want to get latest libraries, you can install development files for
+these formats.
+<br>
+`sudo apt-get install libpng-devel`<br>
+`sudo apt-get install libjpeg-turbo-devel`<br>
+`sudo apt-get install jasper-devel`<br>
+`sudo apt-get install openexr-devel`<br>
+`sudo apt-get install libtiff-devel`<br>
+`sudo apt-get install libwebp-devel`<br>
+
+Several OpenCV functions are parallelized with **Intel's Threading Building Blocks** (TBB). But if
+you want to enable it, you need to install TBB first. ( Also while configuring installation with
+CMake, don't forget to pass -D WITH_TBB=ON. More details below.)
+<br>
+` sudo apt-get install tbb-devel`<br>
+### Downloading OpenCV
+
+Next we have to download OpenCV. You can download the latest release of OpenCV from [sourceforge
+site](http://sourceforge.net/projects/opencvlibrary/). Then extract the folder.
+
+Or you can download latest source from OpenCV's github repo. (If you want to contribute to OpenCV,
+choose this. It always keeps your OpenCV up-to-date). For that, you need to install **Git** first.
```

Comment:
> It always keeps your OpenCV up-to-date

This wording should be improved. Git doesn't do anything automatically, but it helps to to this.